torch/_inductor/codecache.py

```diff
@@ -2910,6 +2914,97 @@ def load(cls, source_code, dst_file_ext) -> Tuple[DLLWrapper, str, str]:
         return (DLLWrapper(dst_file_path), hash_key, source_code_path)
 
 
+@clear_on_fresh_inductor_cache
+class ROCmCodeCache:
```

Comment:
What is different between here and Cuda code cache? _SOURCE_CODE_SUFFIX?

Reply:
Also, importantly, the ROCm implementation should be decoupled from CUDA Toolkit specifics so that they can be independently maintained in the future. Most people who work with CUDA can't test / debug locally if a change breaks the ROCm backend.